src/exporter/DaemonMetricCollector.cc

```diff
@@ -378,6 +373,23 @@ labels_t DaemonMetricCollector::get_extra_labels(std::string daemon_name) {
   return labels;
 }
 
+std::string DaemonMetricCollector::get_instance_id_from_rgw_socket_name(std::string &daemon_socket_file) {
+  if (daemon_socket_file.find('.') != std::string::npos) {
+    // Regular expression to match .<number>.<number> at the end
+    std::regex pattern(R"(\.\d+\.\d+$)");
```

Comment:
I don't think that running regexp in a loop is a good idea. And in terms of backward compatibility, this may break existing alerts/dashboards.

Reply:
> I don't think that running regexp in a loop is a good idea.

Agreed. I'll revisit this.

> And in terms of backward compatibility, this may break existing alerts/dashboards.

Are you saying this because, we are changing the behavior of a label `instance_id` label here instead of introducing a new label. The problem is some of our dashboards are already broken as in the tracker https://tracker.ceph.com/issues/69589 and so, we might need to fix the bug with this label

Reply:
> The problem is some of our dashboards are already broken as in the tracker

the working dashboards will be broken after the update.